website/blog/2025-07-09-full-stack-observability-hyperdx-ferretdb.md

```diff
@@ -0,0 +1,198 @@
+
+- **Open-source:** Both HyperDX and FerretDB are open-source projects, providing transparency, flexibility, and strong community backing, aligning perfectly with an open-source ethos for your entire observability stack.
+- **Simplified infrastructure:** If your existing data infrastructure is already based on PostgreSQL, integrating HyperDX with FerretDB can streamline your database management and reduce operational overhead for your observability solution.
+- **No vendor lock-in:** Enjoy the freedom of truly open-source solutions without concerns about proprietary licensing or vendor lock-in.
+
+## Connecting HyperDX to FerretDB
+
+Connecting a self-hosted HyperDX instance to your FerretDB instance is straightforward, as HyperDX expects a MongoDB-compatible database.
+Here's a step-by-step guide to get you started:
+
+1. **Ensure FerretDB is running:** Make sure your FerretDB instance is active and accessible.
+   You can set up it up with the [FerretDB evaluation image](https://docs.ferretdb.io/installation/evaluation/), by running the following command:
+
+   ```sh
+    docker run -d --name ferretdb -p 27017:27017 \
+    -e POSTGRES_USER=<username> \
+    -e POSTGRES_PASSWORD=<password> \
+    -v ./data:/var/lib/postgresql/data \
+    ghcr.io/ferretdb/ferretdb-eval:2
+   ```
+
+   Replace `<username>` and `<password>` with your desired authentication credentials.
+
+   If you want to explore more installation options, refer to our [FerretDB Installation Guide](https://docs.ferretdb.io/installation/).
+
+2. **Set up HyperDX:** You can run HyperDX as a standalone Docker container or as part of a Docker Compose setup.
+   In this guide, we'll use the recommended Docker Compose setup, which contains HyperDX, an OpenTelemetry Collector, ClickHouse, and other components to facilitate data ingestion and visualization.
+   To deploy HyperDX using Docker Compose, you can clone the HyperDX repository and switch to the `v2` branch, which contains the necessary configurations for self-hosting.
+
+   ```sh
+   git clone https://github.com/hyperdxio/hyperdx.git
+   cd hyperdx
+   # switch to the v2 branch
+   git checkout v2
+   ```
+
+3. **Configure HyperDX's database connection:** Before you start up HyperDX and other components, you'll typically replace the `mongo` service with your ferretdb service.
+   Update the `MONGO_URI` environment variable for HyperDX (with service named `app`) to point to your FerretDB instance.
+
+   ```yaml
+   environment:
+     - MONGO_URI=mongodb://<username>:<password>@ferretdb:27017/hyperdx
+   ```
+
+   Ensure to replace `<username>` and `<password>` with your FerretDB authentication details if enabled and that the container can access your FerretDB instance.
+   Comment out the `mongo` service (set as `db`) in the `docker-compose.yml` file, since you will be using FerretDB instead.
```

Comment:
Also dependence need to be updated https://github.com/hyperdxio/hyperdx/blob/v2/docker-compose.yml#L77.

Just a question, would it be simpler to demonstrate if we provide full docker-compose.yml?

Reply:
Answer to my question?